src/angular/metadata.ts

```diff
@@ -25,39 +25,35 @@ export interface TemplateMetadata extends PropertyMetadata {
 }
 
 export class DirectiveMetadata {
-  constructor(
-    public readonly controller: ts.ClassDeclaration,
-    public readonly decorator: ts.Decorator,
-    public readonly selector?: string
-  ) {}
+  constructor(readonly controller: ts.ClassDeclaration, readonly decorator: ts.Decorator, readonly selector?: string) {}
 }
 
 export class ComponentMetadata extends DirectiveMetadata {
   constructor(
-    public readonly controller: ts.ClassDeclaration,
-    public readonly decorator: ts.Decorator,
-    public readonly selector?: string,
-    public readonly animations?: (AnimationMetadata | undefined)[],
-    public readonly styles?: (StyleMetadata | undefined)[],
-    public readonly template?: TemplateMetadata
+    readonly controller: ts.ClassDeclaration,
+    readonly decorator: ts.Decorator,
+    readonly selector?: string,
+    readonly animations?: (AnimationMetadata | undefined)[],
+    readonly styles?: (StyleMetadata | undefined)[],
+    readonly template?: TemplateMetadata
   ) {
     super(controller, decorator, selector);
   }
 }
 
 export class PipeMetadata {
   constructor(
-    public readonly controller: ts.ClassDeclaration,
-    public readonly decorator: ts.Decorator,
-    public readonly name?: string,
-    public readonly pure?: string
+    readonly controller: ts.ClassDeclaration,
+    readonly decorator: ts.Decorator,
+    readonly name?: string,
```

Comment:
Actually `Angular` doesn't allow you to create a `@Pipe` without `name`...

Reply:
Yes, of course but it's a static analyse.